code/extensions/RealMeSecurityExtension.php

```diff
@@ -91,17 +94,50 @@ public function realme()
      */
     private function realMeACS()
     {
-        $loggedIn = $this->service->enforceLogin();
+        try {
+            $authenticated = $this->service->enforceLogin();
+
+            if($authenticated === true) {
+                $authData = $this->service->getAuthData();
+
+                // If more session vars are set here, they must be cleared in realmeLogout()
+                Session::set('RealMeSessionData', serialize($authData));
+                Session::set('RealMeOriginalResponse', $_POST['SAMLResponse']);
```

Comment:
Fair point, but this is set only after we determine that the user is authenticated via RealMe. So to be malicious, you would have to fake a `SAMLResponse` that is both malicious *and* still passes all the SAML checks (including XML signature validation). So basically, this should be impossible.

Then again, just in case, we can just run `Convert::raw2xml()` over it, as the value saved here is just a base64-encoded string anyway.